packages/lib/src/components/Card/components/CardInput/CardInput.tsx

```diff
@@ -1,330 +1,398 @@
-            prevState.isSfpValid !== this.state.isSfpValid ||
-            prevState.cvcPolicy !== this.state.cvcPolicy ||
-            prevState.hideDateForBrand !== this.state.hideDateForBrand ||
-            prevState.brand !== this.state.brand ||
-            prevState.additionalSelectValue !== this.state.additionalSelectValue ||
-            // Covers changes to:
-            // - encryptedField values
-            // - KCP authentication
-            // - holder name
-            !objectsDeepEqual(prevState.data, this.state.data)
-        ) {
-            this.validateCardInput();
+        if (sfState.autoCompleteName) {
+            if (!props.hasHolderName) return;
+            const holderNameValidationFn = getRuleByNameAndMode('holderName', 'blur');
+            const acHolderName = holderNameValidationFn(sfState.autoCompleteName) ? sfState.autoCompleteName : null;
+            if (acHolderName) {
+                setFormData('holderName', acHolderName);
+                setFormValid('holderName', true); // only if holderName is valid does this fny get called - so we know it's valid and w/o error
+                setFormErrors('holderName', null);
+            }
+            return;
         }
-    }
 
-    public componentWillUnmount() {
-        this.sfp.current.destroy();
-        this.sfp = null;
-    }
+        setData({ ...data, ...sfState.data });
+        setErrors({ ...errors, ...sfState.errors });
+        setValid({ ...valid, ...sfState.valid });
+
+        setIsSfpValid(sfState.isSfpValid);
+
+        // Values relating to /binLookup response
+        setCvcPolicy(sfState.cvcPolicy);
+        setShowSocialSecurityNumber(sfState.showSocialSecurityNumber);
+        setHideDateForBrand(sfState.hideDateForBrand);
+    };
+
+    // Farm the handlers for binLookup related functionality out to another 'extensions' file
+    const extensions = useMemo(
+        () =>
+            CIExtensions(
+                props,
+                { sfp },
+                { dualBrandSelectElements, setDualBrandSelectElements, setSelectedBrandValue, issuingCountryCode, setIssuingCountryCode }
+            ),
+        [dualBrandSelectElements, issuingCountryCode]
+    );
+
+    /**
+     * EXPECTED METHODS ON CARD.THIS
+     */
+    this.showValidation = () => {
+        // Validate SecuredFields
+        sfp.current.showValidation();
 
-    public getChildContext() {
-        return { i18n: this.props.i18n };
-    }
+        // Validates holderName & SSN & KCP (taxNumber)
+        triggerValidation();
 
-    public setStatus(status) {
-        this.setState({ status });
-    }
+        // Validate Address
+        if (billingAddressRef?.current) billingAddressRef.current.showValidation();
+    };
+
+    this.processBinLookupResponse = (binLookupResponse: BinLookupResponse, isReset: boolean) => {
+        extensions.processBinLookup(binLookupResponse, isReset);
+    };
+
+    this.setStatus = status => {
+        setStatus(status);
+    };
```

Comment:
```suggestion
    this.setStatus = setStatus;
```

Very minor, could be simplified to this.